packages/accounts-controller/src/utils.ts

```diff
@@ -85,3 +79,23 @@ export function isNormalKeyringType(keyringType: KeyringTypes): boolean {
   // adapted later on if we have new kind of keyrings!
   return keyringType !== KeyringTypes.snap;
 }
+
+/**
+ * Check if a keyring is a HD keyring.
+ *
+ * @param keyringType - The account's keyring type.
+ * @returns True if the keyring is a HD keyring, false otherwise.
+ */
+export function isHdKeyringType(keyringType: KeyringTypes): boolean {
+  return keyringType === KeyringTypes.hd;
+}
```

Comment:
We could use type predicate here, so we could safely use `HdKeyring` type within a `if` branch:

```ts
export function isHdKeyringType(keyring: EthKeyring): keyring is HdKeyring {
  return keyring.type == KeyringTypes.hd;
}
```

(This would align with my previous comment on using `keyring.hdPath` too)

Reply:
Oh but we are not using a an actual keyring reference (since we get the keyring data from the state directly, this might not work actually 🤔)

Reply:
right.. it does not seem to work as the keyring object we're starting from is not an actual implementation [see my other comment](https://github.com/MetaMask/core/pull/5618#discussion_r2068346806)